lua/nvim-tree.lua

```diff
@@ -205,6 +208,23 @@ local function update_root_dir()
   lib.change_dir(new_cwd)
 end
 
+function M.open_on_directory()
+  local buf = api.nvim_get_current_buf()
+  local bufname = api.nvim_buf_get_name(buf)
+  if vim.fn.isdirectory(bufname) ~= 1 or bufname == lib.Tree.cwd then
+    return
+  end
+
+  if not view.win_open() then
+    view.open()
+  end
+  lib.change_dir(bufname)
+  lib.set_index_and_redraw(bufname)
+  view.focus()
+  vim.cmd ":only"
```

Comment:
This seems like it will break [the "vinegar" use case](https://github.com/tpope/vim-vinegar), which is not to open in "full screen" takeover mode, but rather to open in the current window.

I think the "vinegar" mode is what most of the related issues are actually asking for; only #248 and #214 mention "full-width" display.

Reply:
i think i see what you mean. I did not test that with multiple windows and buffers.
The 'vinegar' use case seems like the right way to do it, i'll implement that soon :)